client/signup/main.jsx

```diff
@@ -346,6 +350,16 @@ class Signup extends Component {
 		}
 	}
 
+	progressBar( flowName ) {
+		this.setState( {
+			progressBar: { flowName, stepName: this.props.stepName },
+		} );
+	}
+
+	isTailoredFlow( flowName ) {
+		this.setState( { isTailoredFlow: [ 'newsletter', 'link-in-bio' ].includes( flowName ) } );
+	}
```

Comment:
These don't need to be stateful, they're both derived from props. They can be deleted.